netket/hilbert/index/constrained.py

```diff
@@ -88,9 +88,11 @@ def _bare_numbers(self) -> np.ndarray:
         return self.__bare_numbers
 
     def states_to_numbers(self, states, out=None):
-        self._unconstrained_index.states_to_numbers(states, out)
+        if out is not None:
+            raise NotImplementedError("Specifying `out`is deprecated.")
+        out = self._unconstrained_index.states_to_numbers(states)
```

Comment:
Sorry, let me rephrase.

Hilbert indices are an internal thing not part of the PUBLIC api.
You can freely remove `out` from here and not give any informative error,
In fact, please do!